zone/tribute.cpp

```diff
@@ -435,9 +386,239 @@ bool ZoneDatabase::LoadTributes() {
+		GuildTributeAbility_Struct* gtas = (GuildTributeAbility_Struct*)outapp->pBuffer;
+
+		auto tier_count = t.second.tier_count;
+		for (int ti = 0; ti < t.second.tier_count; ti++) {
+			if (RuleB(Guild, UseCharacterMaxLevelForGuildTributes) && t.second.tiers[ti].level > RuleI(Character, MaxLevel)) {
+				tier_count -= 1;
+				continue;
+			}
+			gtas->guild_id = GuildID();
+			gtas->ability.tier_count = htonl(tier_count);
+			gtas->ability.tribute_id = htonl(t.first);
+			gtas->ability.tiers[ti].cost = htonl(t.second.tiers[ti].cost);
+			gtas->ability.tiers[ti].tribute_item_id = htonl(t.second.tiers[ti].tribute_item_id);
+			gtas->ability.tiers[ti].level = htonl(t.second.tiers[ti].level);
+		}
+		strcpy(gtas->ability.name, t.second.name.data());
+		FastQueuePacket(&outapp);
+	}
+}
+
+void Client::SendGuildTributeDetails(uint32 tribute_id, uint32 tier)
+{
+	if (tribute_list.count(tribute_id) != 1) {
+		LogError("Details request for invalid tribute [{}]", tribute_id);
+		return;
+	}
+
+	TributeData& td = tribute_list[tribute_id];
+
+	int len = td.description.length();
+	EQApplicationPacket* outapp = new EQApplicationPacket(OP_GuildSelectTribute, sizeof(GuildTributeSelectReply_Struct) + len + 1);
+	GuildTributeSelectReply_Struct* t = (GuildTributeSelectReply_Struct*)outapp->pBuffer;
+
+	t->tribute_id = tribute_id;
+	t->tier = tier;
+	t->tribute_id2 = tribute_id;
+	strncpy(&t->description, td.description.c_str(), len);
+
+	QueuePacket(outapp);
+	safe_delete(outapp);
+}
+
+void Client::DoGuildTributeUpdate()
+{
+	LogTribute("DoGuildTributeUpdate");
+	auto guild = guild_mgr.GetGuildByGuildID(GuildID());
+
+	if (guild && guild->tribute.enabled && GuildTributeOptIn()) {
+		TributeData& d1 = tribute_list[guild->tribute.id_1];
+		uint32 item_id1 = d1.tiers[guild->tribute.id_1_tier].tribute_item_id;
+		TributeData& d2 = tribute_list[guild->tribute.id_2];
+		uint32 item_id2 = d2.tiers[guild->tribute.id_2_tier].tribute_item_id;
+
+		if (item_id1) {
+			LogInfo("Guild Tribute Item 1 is {}", item_id1);
+			const EQ::ItemInstance* inst = database.CreateItem(item_id1, 1);
+			if (inst == nullptr) {
+				LogError("Guild Tribute Item 1 was not found. {}", item_id1);
+				return;
+			}
+			auto inst_level = d1.tiers[guild->tribute.id_1_tier].level;
+			if (m_inv[EQ::invslot::GUILD_TRIBUTE_BEGIN]) {
+				LogInfo("Guild Tribute DELETE Item in Slot 450");
+				DeleteItemInInventory(EQ::invslot::GUILD_TRIBUTE_BEGIN);
+			}
+
+			if ((RuleB(Guild,UseCharacterMaxLevelForGuildTributes) && RuleI(Character, MaxLevel) >= inst_level && GetLevel() >= inst_level) ||
+				!RuleB(Guild, UseCharacterMaxLevelForGuildTributes)) {
+				PutItemInInventory(EQ::invslot::GUILD_TRIBUTE_BEGIN, *inst);
+				SendItemPacket(EQ::invslot::GUILD_TRIBUTE_BEGIN, inst, ItemPacketGuildTribute);
+			}
+
+			safe_delete(inst);
+		}
+
+		if (item_id2) {
+			LogInfo("Guild Tribute Item 2 is {}", item_id2);
```

Comment:
Probably a little debuggy for `LogInfo` perhaps rescope to `LogGuilds` ?

Reply:
Changed to LogGuilds